### ISSUE TITLE
zfs-mount.service: Failed at step EXEC spawning /usr/sbin/zfs: No such file or directory

### ISSUE THREAD

**Issue by ahjohannessen** [varasys/corezfs#6]
I have tried to do installations of zfs using Container Linux ISO and vmware images for 1520.6.0. I have tried to build using corezfs and also tried your latest released archive. However, there seems to be something odd happening because some of zfs-*.service do not start correctly. This is what I get in journald:

```
zfs-mount.service: Failed at step EXEC spawning /usr/sbin/zfs: No such file or directory
zfs-zed.service: Failed at step EXEC spawning /usr/sbin/zed: No such file or directory
zfs-share.service: Failed at step EXEC spawning /usr/sbin/zfs: No such file or directory
```

I can start those manually like this:

```
sudo systemctl start zfs-mount
sudo systemctl start zfs-zed
sudo systemctl start zfs-share
```

Have you seen this issue before?

**Comment by ahjohannessen** [comment]
I got things working changing `zfs-overlay.service` from this:

```
[Unit]
Description=CoreZFS /usr overlay
DefaultDependencies=no
ConditionPathExists=/opt/corezfs/usr
Requires=zfs.target
After=zfs.target
[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/bin/mount -t overlay overlay -o lowerdir=/opt/corezfs/usr:/usr /usr
ExecStartPost=/usr/sbin/modprobe zfs
[Install]
RequiredBy=local-fs.target
```

to this:

```
[Unit]
Description=CoreZFS /usr overlay
DefaultDependencies=no
ConditionPathExists=/opt/corezfs/usr
Before=zfs-import-cache.service
Before=zfs-import-scan.service
Before=zfs-mount.service
Before=zfs-share.service
Before=zfs-zed.service

[Service]
Type=oneshot
RemainAfterExit=yes
ExecStart=/usr/bin/mount -t overlay overlay -o lowerdir=/opt/corezfs/usr:/usr /usr
ExecStartPost=/usr/sbin/modprobe zfs

[Install]
RequiredBy=local-fs.target
```

Not sure what is the permanent fix for this, but using:

```
Requires=zfs.target
After=zfs.target
```

does not seem to be the correct approach, as that is later in the dependency chain than, say, `zfs-mount` and similar. Hence, the `No such file or directory` errors.

**Comment by kcwitt** [comment]
This is fixed in v2.0.2
